e2e/machineset_migration_test.go

```diff
@@ -259,9 +671,10 @@ func createCAPIMachineSet(ctx context.Context, cl client.Client, replicas int32,
 		awsMachineTemplate.Spec.Template.Spec.InstanceType = instanceType
 	}
 
-	if err := cl.Create(ctx, awsMachineTemplate); err != nil {
-		Expect(err).ToNot(HaveOccurred())
-	}
+	Eventually(func() error {
+		return cl.Create(ctx, awsMachineTemplate)
+	}, capiframework.WaitMedium, capiframework.RetryMedium).Should(Succeed(), "Failed to create a new awsMachineTemplate %s",
+		awsMachineTemplate.Name)
```

Comment:
Do we need the func() wrapper here? 

```suggestion
	Eventually(cl.Create(ctx, awsMachineTemplate), capiframework.WaitMedium, capiframework.RetryMedium).Should(Succeed(), "Failed to create a new awsMachineTemplate %s",
		awsMachineTemplate.Name)
```